test/libsolidity/SMTChecker.cpp

```diff
@@ -1,759 +0,0 @@
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Assertion violation happens here");
-	// Check that negation of condition is not assumed after the body anymore
-	text = R"(
-		contract C {
-			function f(uint x) public pure {
-				while (x == 2) {
-				}
-				assert(x != 2);
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Assertion violation happens here");
-	// Check that side-effects of condition are taken into account
-	text = R"(
-		contract C {
-			function f(uint x, uint y) public pure {
-				x = 7;
-				while ((x = y) > 0) {
-				}
-				assert(x == 7);
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Assertion violation happens here");
-}
-
-BOOST_AUTO_TEST_CASE(constant_condition)
-{
-	string text = R"(
-		contract C {
-			function f(uint x) public pure {
-				if (x >= 0) { revert(); }
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Condition is always true");
-	text = R"(
-		contract C {
-			function f(uint x) public pure {
-				if (x >= 10) { if (x < 10) { revert(); } }
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Condition is always false");
-	// a plain literal constant is fine
-	text = R"(
-		contract C {
-			function f(uint) public pure {
-				if (true) { revert(); }
-			}
-		}
-	)";
-	CHECK_SUCCESS_NO_WARNINGS(text);
-}
-
-
-BOOST_AUTO_TEST_CASE(for_loop)
-{
-	string text = R"(
-		contract C {
-			function f(uint x) public pure {
-				require(x == 2);
-				for (;;) {}
-				assert(x == 2);
-			}
-		}
-	)";
-	CHECK_SUCCESS_NO_WARNINGS(text);
-	text = R"(
-		contract C {
-			function f(uint x) public pure {
-				for (; x == 2; ) {
-					assert(x == 2);
-				}
-			}
-		}
-	)";
-	CHECK_SUCCESS_NO_WARNINGS(text);
-	text = R"(
-		contract C {
-			function f(uint x) public pure {
-				for (uint y = 2; x < 10; ) {
-					assert(y == 2);
-				}
-			}
-		}
-	)";
-	CHECK_SUCCESS_NO_WARNINGS(text);
-	text = R"(
-		contract C {
-			function f(uint x) public pure {
-				for (uint y = 2; x < 10; y = 3) {
-					assert(y == 2);
-				}
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Assertion violation");
-	text = R"(
-		contract C {
-			function f(uint x) public pure {
-				uint y;
-				for (y = 2; x < 10; ) {
-					y = 3;
-				}
-				assert(y == 3);
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Assertion violation");
-	text = R"(
-		contract C {
-			function f(uint x) public pure {
-				uint y;
-				for (y = 2; x < 10; ) {
-					y = 3;
-				}
-				assert(y == 2);
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Assertion violation");
-}
-
-BOOST_AUTO_TEST_CASE(division)
-{
-	string text = R"(
-		contract C {
-			function f(uint x, uint y) public pure returns (uint) {
-				return x / y;
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Division by zero");
-	text = R"(
-		contract C {
-			function f(uint x, uint y) public pure returns (uint) {
-				require(y != 0);
-				return x / y;
-			}
-		}
-	)";
-	CHECK_SUCCESS_NO_WARNINGS(text);
-	text = R"(
-		contract C {
-			function f(int x, int y) public pure returns (int) {
-				require(y != 0);
-				return x / y;
-			}
-		}
-	)";
-	CHECK_WARNING(text, "Overflow");
```

Comment:
Ok, the reason why this test didn't fail, resp. behave erratically before was that we didn't even check whether it was ``Overflow ... might happen`` or ``Overflow ... happens`` before.